### DISCUSSION TITLE
Cannot edit files inside devcontainer container in vscode

### DISCUSSION THREAD

**Discussion by takan1** [devcontainers/community#82] [Q&A]
Hi, I successfully created a devcontainer image with `devcontainer build` command with cli. Now I want to use it to edit files inside the docker image so what I did was that I created new folder `.devcontainer` with below files. 
`devcontainer.json`
```
{
    "name": "test docker",
    "dockerFile": "Dockerfile"
}
```
`Dockerfile`
```
FROM my-image:latest
USER myuser
RUN chown -R myuser:mygroup /workspace
WORKDIR /workspace
```
I tried to `reopen in container` in vscode and it started the environment but somehow in a differently set workdir. 
Then I went to the right directory and tried to modify the files but it didn't let me modify it ( cannot save the file). 
This is the Dockerfile that I used to build the image.
```
FROM mcr.microsoft.com/devcontainers/python:1-3.11-bullseye
WORKDIR /workspace
RUN groupadd -r mygroup && useradd -r -g mygroup myuser
RUN chown -R myuser:mygroup /workspace
USER myuser
COPY . .
```
How can I fix this issue?

**Comment by bamurtaugh** [comment]
cc @chrmarti for any thoughts

**Comment by chrmarti** [comment]
We mount the local workspace folder under `/workspaces/your-project` and open that (ignoring `WORKDIR`). You can use `"workspaceMount"` and `"workspaceFolder"` in the devcontainer.json to change this.

If you are running on Linux, we change the UID and GID of the container user to match those of your local user. If this causes issues, you can turn it off with `"updateRemoteUserUID": false` in the devcontainer.json.

**Reply by takan1** [reply]
@chrmarti Thank you so much for the comment! The workspace issue is solved now but I can't still modify and save the files even if I add `"updateRemoteUserUID": false`. I also tried `containerUser` and `remoteUser` with `myuser` as well but it didn't fix it. Btw I'm on M1 Macbook. Any other idea why it's still happening?

**Reply by chrmarti** [reply]
Are you using Docker Desktop or some other Docker installation? Docker Desktop automatically adjusts file ownership.
